pkg/bridge/common.go

```diff
@@ -28,17 +31,83 @@ func sortLogicalBridges(bridges []*pb.LogicalBridge) {
 	})
 }
 
+func (s *Server) createLogicalBridge(lb *pb.LogicalBridge) (*pb.LogicalBridge, error) {
+	// check parameters
+	if err := s.validateLogicalBridgeSpec(lb); err != nil {
+		return nil, err
+	}
+
+	// translation of pb to domain object
+	domainLB := infradb.NewLogicalBridge(lb)
+	// Note: The status of the object will be generated in infraDB operation not here
+	if err := infradb.CreateLB(domainLB); err != nil {
```

Comment:
If I am not mistaken, `CreateLB` will send a msg to task manager, which will find someone to really create a bridge. But why do we return here ok and subsequently from the server's `CreateLogicalBridge`. Should we wait when it is really created? What if creation fails?

Reply:
Here by createLB you create a LogicalBridge intent in the DB. The actual intent is getting realized asychronously by the different components that put the configuration in place. 

It is the same paradigm as in K8s. In K8s you create an intent in the database and then that intent get's realized by the different controllers. It is the same thing here.

If the intent is created in the DB then the call is succesful. We should not wait until the intent get's realized. If an error is occured during the realization of the intent then that is reported asychronously in the status of the object.

Reply:
> If an error is occured during the realization of the intent then that is reported asychronously in the status of the object
How will a client know that? By trying to get an object until it is returned successfully?

Also, our API does not say anything about intents. If a bridge returns ok, it implies that an object was created and functional.
If we introduce a concept of intents, it should be clearly stated in API. But I am not sure if it is a good idea to add intents 🤔

Reply:
so we have API update to indicate that pending state, ok